src/shims/foreign_items.rs

```diff
@@ -110,12 +105,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     /// by this function.
     /// Returns Ok(Some(body)) if processing the foreign item
     /// is delegated to another function.
+    #[rustfmt::skip]
```

Comment:
Oh wow we really have to exclude the entire function for this?

Reply:
Yea, rustfmt has this funny bug where it duplicates the `#[rustfmt::skip]` if I put it on the patterns. I have no clue why

Reply:
It doesn't happen everywhere, since in the other file I can put it on the patterns

Reply:
Oh, weird... we should probably report a bug?

Also would be great if rsutfmt could stop messing with patterns that are already nicely formatted, like one leading `|` per line.

Reply:
I'll open issues right away

Reply:
https://github.com/rust-lang/rustfmt/issues/3973

and

https://github.com/rust-lang/rustfmt/issues/3974